src/EditorFeatures/Test/Snippets/RoslynLSPSnippetConvertTests.cs

```diff
@@ -0,0 +1,243 @@
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.RoslynLSPSnippetConverter)]
+        public Task TestExtendSnippetTextChangeForwardsForCaretThenPlaceholder()
+        {
+            var markup =
+@"[|if (true)
+{
+}|] $$ {|placeholder:test|}";
+
+            var expectedLSPSnippet =
+@"if (true)
+{
+} $0 ${1:test}";
+
+            return TestAsync(markup, expectedLSPSnippet);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.RoslynLSPSnippetConverter)]
+        public Task TestExtendSnippetTextChangeBackwardsForPlaceholderThenCaret()
+        {
+            var markup =
+@"{|placeholder:test|} $$ [|if (true)
+{
+}|]";
+
+            var expectedLSPSnippet =
+@"${1:test} $0 if (true)
+{
+}";
+
+            return TestAsync(markup, expectedLSPSnippet);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.RoslynLSPSnippetConverter)]
+        public Task TestExtendSnippetTextChangeBackwardsForCaretThenPlaceholder()
+        {
+            var markup =
+@"$$ {|placeholder:test|} [|if (true)
+{
+}|]";
+
+            var expectedLSPSnippet =
+@"$0 ${1:test} if (true)
+{
+}";
+
+            return TestAsync(markup, expectedLSPSnippet);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.RoslynLSPSnippetConverter)]
+        public Task TestExtendSnippetTextChangeBackwardsForCaretForwardsForPlaceholder()
+        {
+            var markup =
+@"$$ [|if (true)
+{
+}|] {|placeholder:test|}";
+
+            var expectedLSPSnippet =
+@"$0 if (true)
+{
+} ${1:test}";
+
+            return TestAsync(markup, expectedLSPSnippet);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.RoslynLSPSnippetConverter)]
+        public Task TestExtendSnippetTextChangeBackwardsForPlaceholderForwardsForCaret()
+        {
+            var markup =
+@"{|placeholder:test|} [|if (true)
+{
+}|] $$";
+
+            var expectedLSPSnippet =
+@"${1:test} if (true)
+{
+} $0";
+
+            return TestAsync(markup, expectedLSPSnippet);
+        }
+
+        #endregion
+
+        #region LSP Snippet generation tests
+
+        [Fact, Trait(Traits.Feature, Traits.Features.RoslynLSPSnippetConverter)]
+        public Task TestForLoopSnippet()
+        {
+            var markup =
+@"[|for (var {|placeholder1:i|} = 0; {|placeholder1:i|} < {|placeholder2:length|}; {|placeholder1:i|}++)
+{$$
+}|]";
+
+            var expectedLSPSnippet =
+@"for (var ${1:i} = 0; ${1:i} < ${2:length}; ${1:i}++)
+{$0
+}";
+
+            return TestAsync(markup, expectedLSPSnippet);
+        }
+
+        #endregion
+
+        protected static TestWorkspace CreateWorkspaceFromCode(string code)
+         => TestWorkspace.CreateCSharp(code);
+
+        private static async Task TestAsync(string markup, string output)
+        {
+            MarkupTestFile.GetPositionAndSpans(markup, out var text, out var cursorPosition, out IDictionary<string, ImmutableArray<TextSpan>> placeholderDictionary);
```

Comment:
```suggestion
            MarkupTestFile.GetPositionAndSpans(markup, out var text, out int cursorPosition, out IDictionary<string, ImmutableArray<TextSpan>> placeholderDictionary);
```

Reply:
if I do that then I get an error saying the types are mismatched , which, when fixed, gives a warning to just use var